### ISSUE TITLE
DELETE getting truncated?

### ISSUE THREAD

**Issue by jdlovins** [ErikMinekus/sm-ripext#10]
Oh do i have an interesting bug for you!

```
void DeleteZone(int client, int zoneID) {
	
	//Mark as deleted
	for (int i = 0; i < g_zones.Length; i++) {
		Zone zone = view_as<Zone>(g_zones.Get(i));
		
		if (zone.id == zoneID) {
			zone.id = -1;
			break;
		}
	}
	
	//Send request
	char buffer[128];
	
	Format(buffer, sizeof buffer, "zones/%i/", zoneID);
	
	PrintToServer("Buffer: %s", buffer);
	
	g_timer_client.Delete(buffer, OnDeleteZone);
}
```

Looks pretty normal right? Lets say we give it zoneID of 10

It should call a: DELETE http://api.domain.com/timer/zones/10

Except it just truncates the entire buffer apparently, on my web server logs it actually just calls: DELETE http://api.domain.com/timer/

Like it just lost the buffer somewhere along the way. It's honestly very bizarre. 

Here is the code of the plugin and where it calls the method; https://github.com/mvonsm/timer/blob/development/addons/sourcemod/scripting/timer-map.sp#L788

Also instead of calling DeleteZone() on line 788 and instead put ``g_timer_client.Get("zones/10, OnDeleteZone);`` it works. It seems like it breaks when the call is made in a Void that gets called

I dont understand why it does it, maybe you can compile a version that outputs the URL its going to call for me and we can see if it actually gets lost before CURL calls it or something else.

Edit: so if i do this instead, the calls works. Apparently Format() breaks it?

```
void DeleteZone(int client, int zoneID) {

	//Mark as deleted
	for (int i = 0; i < g_zones.Length; i++) {
		Zone zone = view_as<Zone>(g_zones.Get(i));
		
		if (zone.id == zoneID) {
			zone.id = -1;
			break;	
		}	
	}

	g_timer_client.Delete("zones/10", OnDeleteZone, client);
}
```

**Comment by jdlovins** [comment]
so... I fixed it but theres still a memory issue SOMEWHERE along the way.

```
static char buffer[16];
Format(buffer, sizeof buffer, "zones/%i/", zoneID);
g_timer_client.Delete(buffer, OnDeleteZone);
```
Adding static to the buffer will fix it. 🎉

**Comment by asherkin** [comment]
What version are you running? I think this was fixed in 1.0.2 back in July.

**Comment by jdlovins** [comment]
You know, I'm not exactly sure. I actually cant find rip.ext.so in my extensions folder but my plugin is still working and depends on it. This is spooky
